From b36d25744e2db7989352ad48e00f4c7e87ed65c2 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 24 Dec 2021 05:13:08 +0330
Subject: [PATCH] cmnghttp2: check for HAVE_SIZEOF_SSIZE_T and not HAVE_SSIZE_T

The `check_size_type(ssize_t SIZEOF_SSIZE_T` call in cmcurl (referenced
by the comment above, which also references some other variables that
no longer seem to be used) defines HAVE_SIZEOF_SSIZE_T and not
HAVE_SSIZE_T.
The HAVE_SSIZE_T variable *does* get defined, but via the
`CHECK_SIZE_TYPE(ssize_t SSIZE_T)` call in cmlibarchive, which gets
configured *after* cmnghttp2, and so the first configure leads to an
invalid cmnghttp2/config.h file.
---
 Utilities/cmnghttp2/CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/Utilities/cmnghttp2/CMakeLists.txt b/Utilities/cmnghttp2/CMakeLists.txt
index 3bc2778e..7e8bf99b 100644
--- a/Utilities/cmnghttp2/CMakeLists.txt
+++ b/Utilities/cmnghttp2/CMakeLists.txt
@@ -7,10 +7,8 @@ elseif(CMAKE_C_COMPILER_ID STREQUAL "PathScale")
 endif()
 
 # Re-use some check result cache entries from cmcurl:
-# * HAVE_ARPA_INET_H
-# * HAVE_NETINET_IN_H
-# * HAVE_SSIZE_T
-if(NOT HAVE_SSIZE_T)
+# * HAVE_SIZEOF_SSIZE_T
+if(NOT HAVE_SIZEOF_SSIZE_T)
   set(ssize_t KWIML_INT_intptr_t)
 endif()
 configure_file(cmakeconfig.h.in config.h)
-- 
2.34.1

